### ISSUE TITLE
rumdl fmt applies auto-fixes for disabled rules (MD070)

### ISSUE THREAD

**Issue by seanb4t** [rvben/rumdl#502]
## Summary

`rumdl fmt` applies auto-fix transformations for rules that are disabled in `.rumdl.toml`. Specifically, with `MD070` disabled, `rumdl check` correctly reports no issues, but `rumdl fmt` still inserts closing code fences — introducing MD031 and MD040 violations that weren't there before.

## Reproduction

**`.rumdl.toml`:**
```toml
[global]
disable = ["MD070"]
```

**`test.md`** (a file with nested code fences inside a markdown code block — a documentation file showing code examples):

The file contains patterns like:
````markdown
```bash
some-cli-tool list --format=table
```

```text

```text
````

**Steps:**
```bash
$ rumdl --version
rumdl 0.1.42

$ rumdl check test.md
Success: No issues found in 1 file (13ms)

$ rumdl fmt test.md
Success: No issues found in 1 file (30ms)

$ rumdl check test.md
test.md:216:1: [MD031] No blank line after fenced code block [*]
test.md:217:1: [MD031] No blank line before fenced code block [*]
test.md:217:1: [MD040] Code block (```) missing language [*]
# ... 15 issues total
```

## Expected behavior

`rumdl fmt` should not apply auto-fixes for rules listed in `disable`. If a rule is disabled, both `check` and `fmt` should skip it entirely.

## Actual behavior

`rumdl fmt` applies the MD070 auto-fix (inserting closing ``` fences) even though MD070 is disabled, which corrupts the file and introduces MD031/MD040 violations.

## Impact

This is particularly problematic when `rumdl fmt` runs in CI or as a git hook — it silently corrupts files that previously passed `rumdl check`, causing subsequent lint failures.

## Environment

- rumdl 0.1.42 (Homebrew, macOS arm64)
- Darwin 25.4.0

**Comment by seanb4t** [comment]
Probably shouldn't support nested fences
